### ISSUE TITLE
Merkle

### ISSUE THREAD

**Issue by Col-blimp** [arvidn/libtorrent#1757]
```
assertion failed. Please file a bugreport at https://github.com/arvidn/libtorrent/issues
Please include the following information:

version: 1.2.0.0-62cdec8

#: 1
file: 'src\torrent.cpp'
line: 334
function: libtorrent::torrent::torrent
expression: <unconditional>

stack:
 0: 00007FF796139297 print_backtrace +375 
```

Assertion occurs when adding a torrent created as a merkle torrent but without passing in the merkle tree, I assumed that in the absence of passing in the tree LibTorrent would assume it's downloading and would get what's necessary from peers.

asserts.log when this error is generated is created in the location of the .torrent file not the location of the application .exe, is this expected behaviour?

`settings_pack::support_merkle_torrents` is set to true, when set to false `add_torrent_alert` adds the torrent without the assertion and error is `error_code_enum::no_error` and that looks wrong given merkle is unsupported?

**Comment by arvidn** [comment]
If you have a chance, it would be really helpful if you could write a test that would reproduce this issue (probably by extending ``simulations/test_swarm.cpp``)

**Comment by Col-blimp** [comment]
The simple client and client test produce the error, a sample torrent is here: http://s000.tinyupload.com/index.php?file_id=96868270061824021153

it's just LibTorrent Master created as a merkle torrent, with the default test the tree is not added in but I do get other errors once it's added in and resumed but I think they are related to this issue.

**Comment by arvidn** [comment]
an automated test would be very helpful. I tested merkle torrents a few years ago, and they worked. The reason why they don't anymore is because there's no (automated) test.

**Comment by Col-blimp** [comment]
I shall make one.

**Comment by Col-blimp** [comment]
Out of curiosity is there any ready made visual studio projects for the testing and simulation?

**Comment by arvidn** [comment]
no, and the CMakeLists.txt in the root, I don't think, builds the simulator.
